dali/operators/math/expressions/arithmetic.h

```diff
@@ -84,6 +84,20 @@ inline TileCover GetOneTilePerSample(const TensorListShape<> &shape) {
   return std::make_tuple(std::move(descs), std::move(ranges));
 }
 
+/**
+ * @brief Checks if the child is a (possibly improper) suffix of the parent.
+ * The child must not be longer than the parent.
+ */
+inline bool HasSuffix(const TensorLayout &parent, const TensorLayout &child) {
+  assert(parent.size() >= child.size());
```

Comment:
Unnecessary and actually harmful. Obviously a longer string is not a suffix of a shorter one.
```suggestion
  if (parent.size() < child.size())
    return false;
```

Reply:
done